activerecord/test/cases/insert_all_test.rb

```diff
@@ -787,6 +801,10 @@ def test_insert_all_create_with
     assert_difference "Book.where(format: 'X').count", +2 do
       Book.create_with(format: "X").insert_all!([ { name: "A" }, { name: "B" } ])
     end
+
+    assert_difference "Book.where(name: 'C').count", +2 do
+      Book.create_with(name: "C").insert_all!([ { name: "A" }, { name: "B" } ])
+    end
```

Comment:
This to me feels like a bug in one direction or the other:

- `create_with` should always apply and we should warn/raise if inserts contain the same keys (since those keys will be ignored)
or
- `create_with` should just be a default and specified keys should be able to override it